src/Adapter/Product/Update/ProductShopUpdater.php

```diff
@@ -226,4 +228,27 @@ private function copyImageAssociations(ProductId $productId, ShopId $sourceShopI
             $image->associateTo($targetShopId->getValue(), $productId->getValue());
         }
     }
+
+    /**
+     * @param ProductId $productId
+     * @param ShopId $sourceShopId
+     * @param ShopId $targetShopId
+     *
+     * @return void
+     */
+    private function copyCombinations(ProductId $productId, ShopId $sourceShopId, ShopId $targetShopId, string $productType): void
```

Comment:
I think this should be handled now that this PR https://github.com/PrestaShop/PrestaShop/pull/30683 has been merged, I already fixed combinations copy from to shop to shop (which is why you have conflicts right now)
I didn't notice the delete wasn't handled though, good catch!